a3fe/run/leg.py

```diff
@@ -268,11 +269,12 @@ def setup(
             self.stages.append(
                 _Stage(
                     stage_type=stage_type,
+                    leg_type=self.leg_type,
                     equil_detection=self.equil_detection,
                     runtime_constant=self.runtime_constant,
                     relative_simulation_cost=self.relative_simulation_cost,
                     ensemble_size=self.ensemble_size,
-                    lambda_values=cfg.lambda_values[self.leg_type][stage_type],
+                    lambda_values=self.engine_config.default_lambda_values[self.leg_type][stage_type],
```

Comment:
Please supply the lambda values from self.lam_vals, rather than the default dict.